### PR TITLE
Avoid category and tag override during import

### PR DESCRIPTION
**Pull request by lucatume**
Ticket: https://central.tri.be/issues/74264

This PR taps into the `Tribe_Tracker` class to avoid resetting an imported event local changes pertaining tags and categories to the imported data.

### CODE REVIEW

**Comment by bordoni** [comment]
Can you remove the `common` update from the PR? I mean leave it to @tr1b0t to do it.

File: src/Tribe/API.php [hunk 1]
```diff
@@ -371,6 +371,24 @@ public static function update_event_cost( $event_id ) {
 		}
 
 		/**
+		 * @param int $event_id The event post ID
+		 * @param array $args An array of arguments supported by the `wp_get_object_terms` function.
+		 *
```
Comment (bordoni) [inline src/Tribe/API.php]: Add `@since`

File: src/Tribe/Aggregator/Event.php [hunk 2]
```diff
@@ -333,6 +334,22 @@ public static function preserve_changed_fields( $data ) {
...
+		// reset any modified taxonomy terms
+		$taxonomy_map = array(
+			'post_tag'	 => 'tags',
+			Tribe__Events__Main::TAXONOMY => 'categories',
+		);
+		foreach ( $post_terms as $taxonomy => $terms ) {
+			if ( ! isset( $modified[ $taxonomy ] ) ) {
+				continue;
+			}
+			$tax_key = isset( $taxonomy_map[ $taxonomy ] ) ?
```
Comment (bordoni) [inline src/Tribe/Aggregator/Event.php]: We could use `Tribe__Utils__Array::get()` here to avoid having 3 lines.

File: src/Tribe/API.php (hunk 1 above)
Comment (lucatume) [inline src/Tribe/API.php]: version? 4.5?

Comment (bordoni) [inline src/Tribe/API.php]: Yeah

**Comment by bordoni** [comment]
👍

**Pull request merged**